Polyfills/Console/Source/Console.cpp

```diff
@@ -36,18 +38,64 @@ namespace
 
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
-        std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
-        {
-            if (i > 0)
-            {
-                ss << " ";
+        
+        std::string formattedString = "";
+        if (info.Length() > 0) {
+            formattedString = info[0].ToString().Utf8Value();
+            // check if this string has substitutions or not
+            std::regex toSub("(%[oOs])|(%(\\d*\\.\\d*)?[dif])");
+            std::smatch matches;
+            
+            // for each argument beyond the first (which is the string itself, try to find a substitution string)
+            for (size_t i = 1; i < info.Length(); i++) {
+                Napi::Value v = info[i];
+
+                // check if there's a corresponding match to this argument
+                if (formattedString.find("%") != std::string::npos && std::regex_search(formattedString, matches, toSub)) {
+                    std::string match = matches[0].str();
+                    std::string converted;
+                    // perform proper formatting
+                    if (match == "%o" || match == "%O" || match == "%s") {
+                        // object: for now just turn into [Object object]
+                        converted = v.ToString().Utf8Value();
+                    }
+                    else if (v.IsNumber() && v.ToString().Utf8Value() != "NaN") {
+                        // number formatting
+                        // if we have the float specified, force convert to a float
+                        Napi::Number number = v.ToNumber();
+                        // question: is there a "good" limit here?
+                        char buffer[100];
+                        int size = 100;
+                        int written;
+
+                        // number cases
+                        if (match.find("f") != std::string::npos) {
+                            written = sprintf_s(buffer, size, match.c_str(), number.DoubleValue());
+                        }
+                        else {
+                            written = sprintf_s(buffer, size, match.c_str(), number.Int64Value());
+                        }
+                        converted = std::string(buffer);
+                    }
+                    else {
+                        converted = "NaN";
+                    }
+                    
+                    // replace converted on the original match place
+                    size_t start_pos = formattedString.find(match);
```

Comment:
Why match this again? You should already have the position from the original match?